### DISCUSSION TITLE
Adaptive banner ads

### DISCUSSION THREAD

**Discussion by Carlosps** [admob-plus/admob-plus#308] [Ideas]
As mentioned [here](https://github.com/admob-plus/admob-plus/issues/158), I believe there are people interested in the adaptive banner support.

Who else would be interested in this feature and would also be willing to fund it to have in this plugin?

They seem to be planning to deprecate smart banners as stated here [android migration](https://developers.google.com/admob/android/migration)

**Comment by ratson** [comment]
@Carlosps It is just start deprecating it, will be long before it being removed, so no plan to support it in short term.

**Reply by RonnySchleicher** [reply]
Hello everyone,

when are adaptive banners supported? Is that planned for the near future?

I am currently using Adaptive Banners and have had very good experiences with them.

Now I'd like to go to admob-plus-cordova, but that's something that I really miss.

**Comment by Carlosps** [comment]
@RonnySchleicher and everyone who is interested in this feature. 

Let's fund this feature? I am willing to pay 150 dollars for this. Who else would be able to help?

I have received some emails from Google mentioning Adaptive Banner could have up to 20% better results than the traditional banners.

**Comment by ratson** [comment]
For Cordova users, it can be displayed with
```js
const banner = new admob.BannerAd({
  adUnitId: 'ca-app-pub-3940256099942544/6300978111',
  size: {
    adaptive: "anchored"
  }
})
return banner.show()
```

**Reply by acemagic94** [reply]
What would it take for it to be supported by Ionic users too?
need native ads and also adaptive banners

**Reply by Carlosps** [reply]
Hello @ratson 

I have implemented the adaptive banner as you mentioned, but I noticed the banner got considerably smaller. Is it expected to behave like this?

![admob-plus1](https://user-images.githubusercontent.com/18351773/146460839-ceedb2d4-c9ec-4671-bca7-b2d401f31ece.jpg)
![admob-plus2](https://user-images.githubusercontent.com/18351773/146460854-e7304137-58ef-46fb-bd33-ede8b62e840c.jpg)

**Comment by RonnySchleicher** [comment]
Hello, 

I've implemented it in my latest version and it works fine. Thank you very much!

**Comment by shadiawad** [comment]
![1](https://user-images.githubusercontent.com/24823430/140464953-53f054b4-a69b-4881-a38e-a2b31040b361.png)

**Comment by MartinMuehlner** [comment]
Hello @ratson - not sure if I am asking the right question: Do you have build already the inline adaptive banners from Admob (https://developers.google.com/admob/android/banner/inline-adaptive) - And if yes: how can we implement this into our code - So that we can choose the right position of the banners in the floating content. Thank you for your assistance!

**Reply by MartinMuehlner** [reply]
@ratson where should I put some money to get an answer?

**Comment by harryt2** [comment]
I think maybe the adaptive banners aren't using the height attribute returned from the Admob SDK. Everything seems to be working but I'm having the same problem as @Carlosps, the add is too short.

**Reply by harryt2** [reply]
@ratson My Admob account manager asked me to update to Adaptive Banners by Sept. 25th for what it's worth.

**Reply by Carlosps** [reply]
> I think maybe the adaptive banners aren't using the height attribute returned from the Admob SDK. Everything seems to be working but I'm having the same problem as @Carlosps, the add is too short.

@harryt2 Would you or anyone be able to create a pull request fixing the issue? I would be willing to pay 100$ for this feature.

I have tried to contact @ratson by email, but he seems to be very busy for the last couple of months. So, I believe the PR would be very appreciated.

**Comment by sksk008** [comment]
did anyone find any solution? @ratson

**Comment by jhanzaibkhadim199** [comment]
Any solution for inline-adaptive banner ads for ionic apps ?
